src/parser/shared/modules/spells/SoulInfusion.tsx

```diff
@@ -0,0 +1,35 @@
+import SPELLS from 'common/SPELLS';
+import CoreChanneling from 'parser/shared/modules/Channeling';
+import Events, {RemoveBuffEvent, CastEvent} from 'parser/core/Events';
+import { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import bosses from 'raids/castlenathria/index'
+
+/* Soul Infusion for Sun king's Salvation */
+/* Example Data: https://Wowanalyzer.com/report/g4Pja6pLHnmQtbvk/32-Normal+Sun+King's+Salvation+-+Kill+(10:14)/Pjurbo/standard/timeline */
+
+class SoulInfusion extends CoreChanneling {
+
+  constructor(options: Options) {
+    super(options);
+    
+    const boss = this.owner.boss;
+    this.active = (boss === bosses.bosses["SunKingsSalvation"]) ? true : false;
```

Comment:
Could this be done better?

Reply:
Im sure we could add a BOSSES file to src/game/shadowlands and make that simpler. I cant think of a way to make it any simpler with what is already available.

Reply:
actually you could probably do something like

`this.active = getBossName(this.owner.fight, false) === "Sun King's Salvation"`

not sure that that is any cleaner tho

Reply:
> Im sure we could add a BOSSES file

> i dont think there are many, if any, instances where we have analysis for a specific encounter so i dont think we ever made anything for it.

Yeah, BOSSES file would make this look the cleanest, but if its not going to get used maybe its not worth to implement it.

Reply:
Is there a reason to create a ternary for this? Isn't it enough to simply do ``this.active = (boss === bosses.bosses["SunKingsSalvation"])``?

Reply:
Does `this.active = (boss === bosses.bosses["SunKingsSalvation"]` === `(boss === bosses.bosses["SunKingsSalvation"]) ? true : false;`? Meaning there is no need to do the whole ? : aspect

Reply:
Fixed. Overthinked it for sure.